src/Tests/dotnet-workload-install.Tests/WorkloadGarbageCollectionTests.cs

```diff
@@ -0,0 +1,393 @@
+
+            CreateManifestRecord("testmanifest", "1.0.0", "6.0.100", "6.0.300");
+            CreateManifestRecord("testmanifest", "2.0.0", "6.0.300", "6.0.300");
+            CreateManifestRecord("testmanifest", "3.0.0", "6.0.300", "6.0.300");
+
+            var (installer, getResolver) = GetTestInstaller("6.0.300");
+
+            var packsToKeep = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk),
+                CreatePackInfo("Xamarin.Android.Framework", "8.6.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.6.0.0", WorkloadPackKind.Library)
+            };
+
+            var packsToCollect = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Framework", "8.4.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.4.7.4", WorkloadPackKind.Library)
+            };
+
+            foreach (var pack in packsToKeep.Concat(packsToCollect))
+            {
+                CreateInstalledPack(pack, "6.0.300");
+            }
+
+            // Write workload install record for 6.0.300
+            var workloadsRecordPath = Path.Combine(_dotnetRoot, "metadata", "workloads", "6.0.300", "InstalledWorkloads");
+            Directory.CreateDirectory(workloadsRecordPath);
+            File.Create(Path.Combine(workloadsRecordPath, "xamarin-android-build"));
+
+            installer.GarbageCollect(getResolver);
+
+            foreach (var pack in packsToCollect)
+            {
+                PackShouldExist(pack, false);
+                PackRecord(pack, "6.0.300").Should().NotExist();
+            }
+            foreach (var pack in packsToKeep)
+            {
+                PackShouldExist(pack, true);
+                PackRecord(pack, "6.0.300").Should().Exist();
+            }
+
+            ManifestRecord("testmanifest", "1.0.0", "6.0.100", "6.0.300").Should().NotExist();
+            ManifestRecord("testmanifest", "2.0.0", "6.0.300", "6.0.300").Should().NotExist();
+            ManifestRecord("testmanifest", "3.0.0", "6.0.300", "6.0.300").Should().Exist();
+
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.100", "testmanifest", "1.0.0", "WorkloadManifest.json")).Should().NotExist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "2.0.0", "WorkloadManifest.json")).Should().NotExist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "3.0.0", "WorkloadManifest.json")).Should().Exist();
+        }
+
+        [Fact]
+        public void GarbageCollectManifestsWithInstallState()
+        {
+            CreateMockManifest("testmanifest", "1.0.0", "6.0.100", sourceManifestName: @"Sample2.json");
+            CreateMockManifest("testmanifest", "2.0.0", "6.0.300", sourceManifestName: @"Sample2_v2.json");
```

Comment:
```suggestion
            manifestToKeepVersion = "2.0.0"
            CreateMockManifest("testmanifest", manifestToKeepVersion, "6.0.300", sourceManifestName: @"Sample2_v2.json");
```

Although not important, I would prefer to have version numbers these as variables, so if we ever want to change the number, we only need to change it in once place. That being said, I can see how that might make it a bit harder to read, but it also makes it more clear what each item is testing.

Reply:
There are so many different versions related in different ways that I don't think the test would be likely to still work if we were to change one of them.  I think it would also end up hurting the readability to have so many different variable names.

So instead, to improve the readability of the new tests, I've added a bunch of comments to them and rearranged them a bit.

Reply:
Ah I see, the comments help a lot. I actually misunderstood the test a bit. Thank you!